server/controllers/request.js

```diff
@@ -0,0 +1,73 @@
+const Request = require("../models/Request");
+const User = require("../models/User");
+const asyncHandler = require("express-async-handler");
+
+// @route POST /request
+// @desc create a new pet sitting request
+// @access Private
+exports.makeRequest = asyncHandler(async (req, res, next) => {
+  const body = req.body;
+  const sitter = await User.findById(body.sitterId);
+  if (!sitter) {
+    res.status(404);
+    throw new Error("Sitter doesn't exist");
+  }
+  const newRequest = await Request.create({
+    ownerId: req.user.id,
+    sitterId: body.sitterId,
+    duration: {
+      start: new Date(body.duration.start),
+      end: new Date(body.duration.end),
+    },
+  });
+  if (newRequest) {
+    res.status(200).json({
+      success: {
+        request: newRequest,
+      },
+    });
+  } else {
+    res.status(500);
+    throw new Error("Error creating request");
+  }
+});
+
+// @route GET /request
+// @desc Get requests for a logged in user
+// @access Private
+exports.getRequest = asyncHandler(async (req, res, next) => {
+  const requests = await Request.find({ sitterId: req.user.id });
+  res.status(200).json({
+    success: {
+      requests: requests,
+    },
+  });
+});
+
+// @route PATCH /request
+// @desc Update a user's request
+// @access Private
+exports.editRequest = asyncHandler(async (req, res, next) => {
+  const body = req.body;
```

Comment:
Validate the request body here. I personally prefer to use a dynamic route for updating a resource. Since `Request` is a resource, you could just send this `id` in the request params.

Reply:
Validation added + route updated to dynamic route. 
It makes sense to not use a body if its not needed. As we are already using the URL we might as well take full advantage of it. I will look to implement this moving forward. Thanks